model/bootstrap/node_info.go

```diff
@@ -177,16 +180,63 @@ type decodableNodeInfoPub struct {
 	Stake uint64
 }
 
+func (info *NodeInfoPriv) MarshalJSON() ([]byte, error) {
+	enc := encodableNodeInfoPriv{
+		Role:           info.role,
+		Address:        info.Address,
+		NodeID:         info.nodeID,
+		NetworkPrivKey: info.NetworkPrivKey,
+		StakingPrivKey: info.StakingPrivKey,
+	}
+	return json.Marshal(enc)
+}
+
+func (info *NodeInfoPriv) UnmarshalJSON(b []byte) error {
+	var dec encodableNodeInfoPriv
+	err := json.Unmarshal(b, &dec)
+	if err != nil {
+		return fmt.Errorf("could not decode json: %w", err)
+	}
+	info.role = dec.Role
+	info.Address = dec.Address
+	info.nodeID = dec.NodeID
+	info.NetworkPrivKey = dec.NetworkPrivKey
+	info.StakingPrivKey = dec.StakingPrivKey
+	return nil
+}
+
 func (info *NodeInfoPub) Equals(other *NodeInfoPub) bool {
 	if other == nil {
 		return false
 	}
 	return info.Address == other.Address &&
-		info.NodeID == other.NodeID &&
-		info.Role == other.Role &&
+		info.nodeID == other.nodeID &&
+		info.role == other.role &&
 		info.Weight == other.Weight &&
 		info.NetworkPubKey.PublicKey.Equals(other.NetworkPubKey.PublicKey) &&
-		info.StakingPubKey.PublicKey.Equals(other.StakingPubKey.PublicKey)
+		info.StakingPubKey.PublicKey.Equals(other.StakingPubKey.PublicKey) &&
+		slices.Equal(info.StakingPoP.Signature, other.StakingPoP.Signature)
+}
+
+func (info *NodeInfoPub) MarshalJSON() ([]byte, error) {
+	enc := struct {
+		Role          flow.Role // role is public
+		Address       string
+		NodeID        flow.Identifier // node ID is public
+		Weight        uint64
+		NetworkPubKey encodable.NetworkPubKey
+		StakingPubKey encodable.StakingPubKey
+		StakingPoP    encodable.StakingKeyPoP
+	}{
```

Comment:
we have the `decodableNodeInfoPub` struct. I think it would help expressiveness if:
1. we implement 
   ```go
   type decodableNodeInfoPub struct {
      encodableNodeInfoPub

      // Stake previously was used in place of the Weight field.
      // Deprecated: supported in decoding for backward-compatibility
      Stake uint64
   }
```
and then use `encodableNodeInfoPub` here